File: RailgunVB/Commands/Root/RootBotlog.vb
Imports Discord
Imports Discord.Commands
Imports RailgunVB.Core.Configuration
Imports RailgunVB.Core.Logging

Namespace Commands.Root

    Partial Public Class Root
    
        <Group("botlog")>
        Public Class RootBotlog
            Inherits  ModuleBase
        
            Private ReadOnly _config As MasterConfig

            Public Sub New(config As MasterConfig)
                _config = config
            End Sub
        
            <Command>
            Public Async Function DefaultAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.Common)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the Default botlog channel.")
            End Function
            
            <Command("audiochord")>
            Public Async Function AudioAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.AudioChord)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the AudioChord botlog channel.")
            End Function
            
            <Command("cmdmngr")>
            Public Async Function CommandAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.CommandManager)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the Command Manager botlog channel.")
            End Function
        
            <Command("guildmngr")>
            Public Async Function GuildAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.GuildManager)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the Guild Manager botlog channel.")
            End Function
            
            <Command("musicmngr")>
            Public Async Function MusicAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.MusicManager)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the Music Manager botlog channel.")
            End Function
            
            <Command("musicplayer")>
            Public Async Function PlayerAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.MusicPlayer)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the Music Player botlog channel.")
            End Function
            
            <Command("tasksch")>
            Public Async Function TaskAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.TaskScheduler)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the Task Scheduler botlog channel.")
            End Function
            
            <Command("timermngr")>
            Public Async Function TimerAsync() As Task
                await _config.AssignBotLogChannelAsync(Context.Channel.Id, BotLogType.TimerManager)
                await ReplyAsync(
                    $"This channel {Format.Bold(Context.Channel.Name)} has been set as the Timer Manager botlog channel.")
            End Function
            
        End Class
    
    End Class
    
End NameSpace

File: Railgun/Commands/Root/RootBotLog.cs
using System;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Discord;
using Finite.Commands;
using Railgun.Core;
using Railgun.Core.Configuration;
using Railgun.Core.Enums;

namespace Railgun.Commands.Root
{
    public partial class Root 
    {
        [Alias("botlog")]
        public class RootBotLog : SystemBase
        {
            private readonly MasterConfig _config;

            public RootBotLog(MasterConfig config)
                => _config = config;

            [Command]
            public Task ExecuteAsync(string logType, ITextChannel textChannel)
            {
                Enum.TryParse(logType, out BotLogType botLogType);

                _config.AssignBotLogChannel(textChannel.Id, botLogType);

                return ReplyAsync(string.Format("The channel {0} has been set as the {1} botlog channel.", 
                    textChannel.Mention, 
                    botLogType));
            }

            [Command]
            public Task ExecuteAsync(string logType)
                => ExecuteAsync(logType, Context.Channel as ITextChannel);

            [Command]
            public Task ExecuteAsync()
            {
                var values = Enum.GetValues(typeof(BotLogType))
                    .Cast<BotLogType>();

                var output = new StringBuilder()
                    .AppendLine("All Bot Log Channels:")
                    .AppendLine();

                foreach (var value in values)
                    output.AppendLine(value.ToString());

                return ReplyAsync(output.ToString());
            }
        }
    }
}